File: app/src/main/java/com/ndmsystems/coala/layers/SendLayer.java
package com.ndmsystems.coala.layers;

import com.ndmsystems.coala.message.CoAPMessage;
import com.ndmsystems.coala.utils.Reference;

import java.net.InetSocketAddress;

public interface SendLayer {
    boolean onSend(CoAPMessage message, Reference<InetSocketAddress> receiverAddress);
}

File: app/src/main/java/com/ndmsystems/coala/layers/SendLayer.kt
package com.ndmsystems.coala.layers

import com.ndmsystems.coala.LayersStack
import com.ndmsystems.coala.message.CoAPMessage
import com.ndmsystems.coala.utils.Reference
import java.net.InetSocketAddress

interface SendLayer {
    fun onSend(message: CoAPMessage, receiverAddressReference: Reference<InetSocketAddress>): LayersStack.LayerResult
}